agents-run-api/src/agents/ModelFactory.ts

```diff
@@ -27,6 +30,11 @@ export class ModelFactory {
         return createOpenAI(config);
       case 'google':
         return createGoogleGenerativeAI(config);
+      case 'openrouter':
+        // Use official OpenRouter provider, cast to handle type mismatch
+        return createOpenRouter(config) as unknown as Provider;
```

Comment:
double cast here seems wrong.

`as unknown as Provider`

Is there a better way to clean up these types?